src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptionsSetup.cs

```diff
@@ -52,8 +57,47 @@ public void Configure(LanguageWorkerOptions options)
                 return;
             }
 
-            var configFactory = new RpcWorkerConfigFactory(_configuration, _logger, SystemRuntimeInformation.Instance, _environment, _metricsLogger, _workerProfileManager);
+            // Use the latest configuration from the ScriptHostManager if available.
+            // After specialization, the ScriptHostManager will have the latest IConfiguration reflecting additional configuration entries added during specialization.
+            var configuration = _configuration;
+            if (_scriptHostManager is IServiceProvider scriptHostManagerServiceProvider)
+            {
+                var latestConfiguration = scriptHostManagerServiceProvider.GetService<IConfiguration>();
+                if (latestConfiguration is not null)
+                {
+                    configuration = new ConfigurationBuilder()
+                        .AddConfiguration(_configuration)
+                        .AddConfiguration(latestConfiguration)
+                        .Build();
+                }
+            }
```

Comment:
Is it possible that this runs once *before* we have the JobHost built, meaning the _scriptHostManager doesn't have an ActiveHost yet, therefore this throws when calling `GetService<T>()`? I just ran into this in some tests, so it's fresh on my mind...

Reply:
In my testing,  When running on WebHost level, there is a ScriptHostManager instance. The `ActiveHost` property on that instance is `null`. The `scriptHostManagerServiceProvider.GetService<IConfiguration>()` returns `null` and our null check handles it.